js_modules/dagster-ui/packages/ui-core/src/assets/RecentUpdatesTimeline.tsx

```diff
@@ -209,29 +209,32 @@ export const RecentUpdatesTimeline = ({assetKey, events, loading}: Props) => {
                   position="top"
                   interactionKind="hover"
                   content={
-                    <Box flex={{direction: 'column', gap: 8}}>
-                      <Box padding={8} border="bottom">
+                    <div style={{width: 400}}>
+                      <Box padding={{vertical: 8, horizontal: 12}} border="bottom">
                         <Subtitle2>Updates</Subtitle2>
                       </Box>
-                      <div style={{maxHeight: 'min(80vh, 300px)', overflow: 'scroll'}}>
-                        {bucket.events
+                      <Box style={{maxHeight: '300px', overflowY: 'auto'}}>
+                        {[...bucket.events]
                           .sort((a, b) => parseInt(b.timestamp) - parseInt(a.timestamp))
                           .map((event, index) => (
-                            <AssetUpdate assetKey={assetKey} event={event} key={index} />
+                            <AssetUpdate
+                              assetKey={assetKey}
+                              event={event}
+                              key={event.timestamp}
```

Comment:
I wonder if it's possible for two events to occur at the same exact timestamp? Probably not? 😬  could make this a composite of timestamp + index just in case